pkg/controller/ec2/natgateway/controller.go

```diff
@@ -135,17 +135,28 @@ func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.Ex
 		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
 	}
 
-	nat, err := e.client.CreateNatGateway(ctx, &awsec2.CreateNatGatewayInput{
-		ConnectivityType: awsec2types.ConnectivityType(cr.Spec.ForProvider.ConnectivityType),
-		AllocationId:     cr.Spec.ForProvider.AllocationID,
-		SubnetId:         cr.Spec.ForProvider.SubnetID,
-		TagSpecifications: []awsec2types.TagSpecification{
-			{
-				ResourceType: "natgateway",
-				Tags:         v1beta1.GenerateEC2Tags(cr.Spec.ForProvider.Tags),
+	var input *awsec2.CreateNatGatewayInput
+	if cr.Spec.ForProvider.Tags != nil {
+		input = &awsec2.CreateNatGatewayInput{
+			ConnectivityType: awsec2types.ConnectivityType(cr.Spec.ForProvider.ConnectivityType),
+			AllocationId:     cr.Spec.ForProvider.AllocationID,
+			SubnetId:         cr.Spec.ForProvider.SubnetID,
+			TagSpecifications: []awsec2types.TagSpecification{
+				{
+					ResourceType: "natgateway",
+					Tags:         v1beta1.GenerateEC2Tags(cr.Spec.ForProvider.Tags),
+				},
 			},
-		},
-	})
+		}
+	} else {
+		input = &awsec2.CreateNatGatewayInput{
+			ConnectivityType: awsec2types.ConnectivityType(cr.Spec.ForProvider.ConnectivityType),
+			AllocationId:     cr.Spec.ForProvider.AllocationID,
+			SubnetId:         cr.Spec.ForProvider.SubnetID,
+		}
+	}
+
```

Comment:
Style nit: I would prefer:

```go
// Create an input without tags.
input := &awsec2.CreateNatGatewayInput{
	ConnectivityType: awsec2types.ConnectivityType(cr.Spec.ForProvider.ConnectivityType),
	AllocationId:     cr.Spec.ForProvider.AllocationID,
	SubnetId:         cr.Spec.ForProvider.SubnetID,
}

// If we specified tags, update the above input.
if cr.Spec.ForProvider.Tags != nil {
	input.TagSpecifications = []awsec2types.TagSpecification{{
		ResourceType: "natgateway",
		Tags:         v1beta1.GenerateEC2Tags(cr.Spec.ForProvider.Tags),
	}}
}
```

At a glance it seems like this should work.

Reply:
thanks for review - tested again and changed the code - works like expected